### PR TITLE
Entities search

### PR DESCRIPTION
**Pull request by kelvin-muchiri**
### Changes / Features implemented

Add search filter support for Entities

### Steps taken to verify this change does what is intended

- [x] QA

**Before submitting this PR for review, please make sure you have:**

  - [x] Included tests
  - [x] Updated documentation

### CODE REVIEW

**Comment by faith-mutua** [comment]
@kelvin-muchiri this is working as expected 
Screenshot from 2024-07-11 10-28-44

File: onadata/apps/api/viewsets/entity_list_viewset.py [hunk 1]
```diff
@@ -149,3 +146,18 @@ def create(self, request, *args, **kwargs):
...
+        search_param = api_settings.SEARCH_PARAM
+        search = request.query_params.get(search_param, "")
+        queryset = Entity.objects.filter(
+            entity_list_id=entity_list.pk, deleted_at__isnull=True
+        )
+
+        if search:
+            queryset = queryset.filter(Q(json__iregex=search) | Q(uuid=search))
+
+        queryset = queryset.order_by("id")
```
Comment (ukanga) [inline onadata/apps/api/viewsets/entity_list_viewset.py:161]: Is the `order by` necessary? Is it likely to be a performance bottleneck?

**Review by ukanga — APPROVED** [review]

File: onadata/apps/api/viewsets/entity_list_viewset.py (hunk 1 above)
Comment (kelvin-muchiri) [reply onadata/apps/api/viewsets/entity_list_viewset.py:161]: @ukanga We can choose not to order by default but it ill still be needed if pagination is applied

Comment (ciremusyoka) [reply onadata/apps/api/viewsets/entity_list_viewset.py:161]: @kelvin-muchiri we fetch this data paginated

**Pull request merged**